test/reg-test/fam-api-reg/fam_control_path_reg_test.cpp

```diff
@@ -0,0 +1,209 @@
+        sprintf(itemInfo, "%s_%d", firstItemLocal, i);
+        // Allocating data items
+        EXPECT_NO_THROW(itemLocal[i] = my_fam->fam_allocate(
+                            itemInfo, DATA_ITEM_SIZE, 0777, descLocal));
+        EXPECT_NE((void *)NULL, itemLocal[i]);
+    }
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+    for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+        EXPECT_NO_THROW(my_fam->fam_deallocate(itemLocal[i]));
+    }
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+    EXPECT_NO_THROW(my_fam->fam_destroy_region(descLocal));
+    free((void *)testRegionLocal);
+    free((void *)firstItemLocal);
+}
+
+TEST(FamDataPath, FamResizeRegion) {
+ Fam_Region_Descriptor *descLocal;
+ const char *testRegionLocal = get_uniq_str("testLocal", my_fam);
+ EXPECT_NO_THROW(descLocal = my_fam->fam_create_region(
+                     testRegionLocal, RESIZE_REGION_SIZE, 0777, RAID1));
+ EXPECT_NE((void *)NULL, descLocal);
+
+ EXPECT_NO_THROW(my_fam->fam_barrier_all());
+ Fam_Stat *info = (Fam_Stat *)malloc(sizeof(Fam_Stat));
+ for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+     EXPECT_NO_THROW(my_fam->fam_stat(descLocal, info));
+     EXPECT_NO_THROW(my_fam->fam_resize_region(descLocal, info->size * i));
+ }
+ free(info);
+ EXPECT_NO_THROW(my_fam->fam_barrier_all());
+ // EXPECT_NO_THROW(rpc->generate_profile());
+ EXPECT_NO_THROW(my_fam->fam_destroy_region(descLocal));
+}
+
+// Test case -  DataItem Change Permission.
+TEST(FamDataPath, DataItemChangePermission) {
+ Fam_Region_Descriptor *descLocal;
+ Fam_Descriptor *itemLocal;
+ const char *testRegionLocal = get_uniq_str("testLocal", my_fam);
+ const char *firstItemLocal = get_uniq_str("firstLocal", my_fam);
+ EXPECT_NO_THROW(descLocal = my_fam->fam_create_region(
+                     testRegionLocal, BIG_REGION_SIZE, 0777, RAID1));
+ EXPECT_NE((void *)NULL, descLocal);
+
+ // Allocating data items
+ EXPECT_NO_THROW(itemLocal = my_fam->fam_allocate(
+                     firstItemLocal, DATA_ITEM_SIZE, 0777, descLocal));
+ EXPECT_NE((void *)NULL, itemLocal);
+ mode_t perm;
+ EXPECT_NO_THROW(my_fam->fam_barrier_all());
+ for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+     perm = (mode_t)i;
+     my_fam->fam_change_permissions(itemLocal, perm);
+ }
+ EXPECT_NO_THROW(my_fam->fam_barrier_all());
+
+ // EXPECT_NO_THROW(rpc->generate_profile());
+ EXPECT_NO_THROW(my_fam->fam_deallocate(itemLocal));
+ EXPECT_NO_THROW(my_fam->fam_destroy_region(descLocal));
+
+ free((void *)testRegionLocal);
+ free((void *)firstItemLocal);
+}
+
+// Test case -  Region Change Permission.
+TEST(FamDataPath, RegionChangePermission) {
+ Fam_Region_Descriptor *descLocal;
+ const char *testRegionLocal = get_uniq_str("testLocal", my_fam);
+ EXPECT_NO_THROW(descLocal = my_fam->fam_create_region(
+                     testRegionLocal, BIG_REGION_SIZE, 0777, RAID1));
+ EXPECT_NE((void *)NULL, descLocal);
+
+ EXPECT_NO_THROW(my_fam->fam_barrier_all());
+ mode_t perm;
+ for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
```

Comment:
Same here - The loop around change permission in microbenchmark was to take performance measurement. We do not need that loop here.